### PR TITLE
python testing: Add PICS parser for xml

### PR DESCRIPTION
**Pull request by cecille**
Code blatently stolen from Jon Rhees and Stefan Bauer-Schwan

### CODE REVIEW

File: src/python_testing/matter_testing_support.py [hunk 1]
```diff
@@ -154,11 +156,32 @@ def parse_pics(lines=typing.List[str]) -> dict[str, bool]:
...
+
+def read_pics_from_file(path: str) -> dict[str, bool]:
+    """ Reads a dictionary of PICS from a file (ci format) or directory (xml format). """
+    if os.path.isdir(os.path.abspath(path)):
+        pics_dict = {}
+        for filename in glob.glob(f'{path}/*.xml'):
+            with open(filename, 'r') as f:
+                contents = f.read()
+                pics_dict.update(parse_pics_xml(contents))
+        print(pics_dict)
```
Comment (andy31415) [inline src/python_testing/matter_testing_support.py:177]: do we need this print here?

File: src/python_testing/matter_testing_support.py [hunk 2]
```diff
@@ -154,11 +156,32 @@ def parse_pics(lines=typing.List[str]) -> dict[str, bool]:
...
+        return pics_dict
+
+    else:
+        print("notdir")
```
Comment (andy31415) [inline src/python_testing/matter_testing_support.py:181]: same here - is this debug code?

**Review by andy31415 — APPROVED** [review]

**Review by raju-apple — APPROVED** [review]

File: src/python_testing/matter_testing_support.py (hunk 2 above)
Comment (cecille) [reply src/python_testing/matter_testing_support.py]: yup - now you've seen my shameful hacks :) Removed.

File: src/python_testing/matter_testing_support.py (hunk 1 above)
Comment (cecille) [reply src/python_testing/matter_testing_support.py]: nope - thanks.

**Pull request merged**